pyaedt/edb_grpc/core/edb_data/padstacks_data.py

```diff
@@ -0,0 +1,1656 @@
+        -------
+        str
+            Name of the stopping layer.
+        """
+        layer = self._pedb.edb.Cell.Layer("", self._pedb.edb.Cell.LayerType.SignalLayer)
+        _, start_layer, stop_layer = self._edb_padstackinstance.GetLayerRange()
+
+        if stop_layer:
+            return stop_layer.GetName()
+        return None
+
+    @stop_layer.setter
+    def stop_layer(self, layer_name):
+        start_layer = self._pedb.stackup.signal_layers[self.start_layer]._edb_layer
+        layer = self._pedb.stackup.signal_layers[layer_name]._edb_layer
+        self._edb_padstackinstance.SetLayerRange(start_layer, layer)
+
+    @property
+    def layer_range_names(self):
+        """List of all layers to which the padstack instance belongs."""
+        _, start_layer, stop_layer = self._edb_padstackinstance.GetLayerRange()
+        started = False
+        layer_list = []
+        start_layer_name = start_layer.GetName()
+        stop_layer_name = stop_layer.GetName()
+        for layer_name in list(self._pedb.stackup.layers.keys()):
+            if started:
+                layer_list.append(layer_name)
+                if layer_name == stop_layer_name or layer_name == start_layer_name:
+                    break
+            elif layer_name == start_layer_name:
+                started = True
+                layer_list.append(layer_name)
+                if layer_name == stop_layer_name:
+                    break
+            elif layer_name == stop_layer_name:
+                started = True
+                layer_list.append(layer_name)
+                if layer_name == start_layer_name:
+                    break
+        return layer_list
+
+    @property
+    def net_name(self):
+        """Net name.
+
+        Returns
+        -------
+        str
+            Name of the net.
+        """
+        return self._edb_padstackinstance.GetNet().GetName()
+
+    @net_name.setter
+    def net_name(self, val):
+        if not isinstance(val, str):
+            try:
+                self._edb_padstackinstance.SetNet(val)
+            except:
+                raise AttributeError("Value inserted not found. Input has to be net name or net object.")
+        elif val in self._pedb.core_nets.nets:
+            net = self._pedb.core_nets.nets[val].net_object
+            self._edb_padstackinstance.SetNet(net)
+        else:
+            raise AttributeError("Value inserted not found. Input has to be net name or net object.")
+
+    @property
+    def is_pin(self):
+        """Determines whether this padstack instance is a layout pin.
+
+        Returns
+        -------
+        bool
+            True if this padstack type is a layout pin, False otherwise.
+        """
+        return self._edb_padstackinstance.IsLayoutPin()
+
+    @is_pin.setter
+    def is_pin(self, pin):
+        """Set padstack type
+
+        Parameters
+        ----------
+        pin : bool
+            True if set this padstack instance as pin, False otherwise
```

Comment:
```suggestion
        """Set the padstack type.

        Parameters
        ----------
        pin : bool
            Whether this padstack instance is a pin. Set ``True`` if it is a pin and
            ``False`` otherwise.
```